tests/ui/test_ui_tools.py

```diff
@@ -409,22 +409,26 @@ def test_init(self, mocker, stream_view):
         self.stream_search_box.assert_called_once_with(
             stream_view, 'SEARCH_STREAMS', stream_view.update_streams)
 
-    @pytest.mark.parametrize('new_text, expected_log', [
-        ('f', ['FOO', 'FOOBAR', 'foo', 'fan']),
-        ('a', ['FOOBAR', 'fan', 'bar']),
-        ('bar', ['FOOBAR', 'bar']),
-        ('foo', ['FOO', 'FOOBAR', 'foo']),
-        ('FOO', ['FOO', 'FOOBAR', 'foo']),
-        ('test', ['test here']),
-        ('here', ['test here']),
+    @pytest.mark.parametrize('new_text, expected_log, to_pin', [
+        ('f', ['FOO', 'FOOBAR', 'foo', 'fan'], []),
+        ('bar', ['bar'], []),
+        ('foo', ['FOO', 'FOOBAR', 'foo'], []),
+        ('FOO', ['FOO', 'FOOBAR', 'foo'], []),
+        ('test', ['test here'], []),
+        ('here', ['test here'], []),
+        ('test here', ['test here'], []),
+        # With 'foo' pinned.
+        ('f', ['foo', 'FOO', 'FOOBAR', 'fan'], [['foo'], ]),
+        ('FOO', ['foo', 'FOO', 'FOOBAR'], [['foo'], ]),
```

Comment:
These test cases are OK, but it feels like duplication of the other test parameters, and don't go into as much depth - and we could also test `match_stream`, fundamentally, perhaps instead of these two.